pallets/subtensor/src/lib.rs

```diff
@@ -1301,21 +1334,29 @@ pub mod pallet {
         // 	* 'n': (BlockNumberFor<T>):
         // 		- The number of the block we are initializing.
         fn on_initialize(_block_number: BlockNumberFor<T>) -> Weight {
+            // Unstake all and transfer pending coldkeys
+            let swap_weight = match Self::swap_coldkeys_this_block() {
```

Comment:
We do need to be careful about how long this operation will take on average, otherwise it may brick the chain.

Reply:
33k IIRC

Reply:
We require caller to submit proofs of work which grow expotentially with each call. In addition, the calling coldkey has to have 1 TAO , which makes DDOS this function very expensive

Reply:
Wait, how does the caller factor in here? This `swap_coldkeys_this_block` function is being run in `on_initialize`, which doesn't have a caller and gets run on every block.

Reply:
in the previous call , `schedule_coldkey_swap` , it registers a arbitration block number, and then this call with swap those keys once they reach the arbitration block number.